adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,712 @@
+static void tlsio_openssl_DestroyOption(const char* name, const void* value)
+{
+    /*since all options for this layer are actually string copies, disposing of one is just calling free*/
+    if (
+        (name == NULL) || (value == NULL)
+        )
+    {
+        LogError("invalid parameter detected: const char* name=%p, const void* value=%p", name, value);
+    }
+    else
+    {
+        if (
+            (strcmp(name, "TrustedCerts") == 0) ||
+            (strcmp(name, "x509certificate") == 0) ||
+            (strcmp(name, "x509privatekey") == 0)
+        )
+        {
+            free((void*)value);
+        }
+        else
+        {
+            LogError("not handled option : %s", name);
+        }
+    }
+}
+
+/* Codes_SRS_TLSIO_SSL_ESP8266_99_001: [ The tlsio_openssl_retrieveoptions shall not do anything, and return NULL. ]*/
+static OPTIONHANDLER_HANDLE tlsio_openssl_retrieveoptions(CONCRETE_IO_HANDLE handle)
+{
+    OPTIONHANDLER_HANDLE result = NULL;
+    return result;
+}
+
+static const IO_INTERFACE_DESCRIPTION tlsio_openssl_interface_description =
+{
+    tlsio_openssl_retrieveoptions,
+    tlsio_openssl_create,
+    tlsio_openssl_destroy,
+    tlsio_openssl_open,
+    tlsio_openssl_close,
+    tlsio_openssl_send,
+    tlsio_openssl_dowork,
+    tlsio_openssl_setoption
+};
+
+static LOCK_HANDLE * openssl_locks = NULL;
+
+static struct CRYPTO_dynlock_value* openssl_dynamic_locks_create_cb(const char* file, int line)
+{
+    struct CRYPTO_dynlock_value* result;
+    return result;
+}
+
+static void openssl_dynamic_locks_uninstall(void)
+{
+}
+
+static void openssl_dynamic_locks_install(void)
+{
+}
+
+static void openssl_static_locks_lock_unlock_cb(int lock_mode, int lock_index, const char * file, int line)
+{
+}
+
+static void indicate_open_complete(TLS_IO_INSTANCE* tls_io_instance, IO_OPEN_RESULT open_result)
+{
+    if (tls_io_instance->on_io_open_complete == NULL)
+    {
+        LogError("NULL on_io_open_complete.");
+    }
+    else
+    {
+        tls_io_instance->on_io_open_complete(tls_io_instance->on_io_open_complete_context, open_result);
+    }
+}
+
+
+static int lwip_net_errno(int fd)
+{
+    int sock_errno = 0;
+    u32_t optlen = sizeof(sock_errno);
+    getsockopt(fd, SOL_SOCKET, SO_ERROR, &sock_errno, &optlen);
+    return sock_errno;
+}
+
+static void lwip_set_non_block(int fd) 
+{
+  int flags = -1;
+  int error = 0;
+
+  while(1){
+      flags = fcntl(fd, F_GETFL, 0);
+      if (flags == -1){
+          error = lwip_net_errno(fd);
+          if (error != EINTR){
+              break;
+          }
+      } else{
+          break;
+      }
+  }
+
+  while(1){
+      flags = fcntl(fd, F_SETFL, flags | O_NONBLOCK);
+      if (flags == -1) {
+          error = lwip_net_errno(fd);
+          if (error != EINTR){
+              break;
+          }
+      } else{
+          break;
+      }
+  }
+
+}
+
+
+LOCAL void openssl_thread_LWIP_CONNECTION(void *p)
+{
+    int ret = -1;
+
+    int sock = -1;
```

Comment:
Please do not initialize these variables here with a 'predict' value. It is better to have a compilation error if you try to use it without initialization than introduce a bug in the execution time because you are using it with a wrong value.